### PR TITLE
Update template writing guide for best user experience

### PR DESCRIPTION
**Pull request by jwforres**
Implements https://trello.com/c/LBm5DB8b

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett @bparees @luciddreamz @sspeiche

**Comment by jwforres** [comment]
Updates the docs page here: https://docs.openshift.org/latest/dev_guide/templates.html#writing-description

template-part-1


And here: https://docs.openshift.org/latest/dev_guide/templates.html#writing-parameters

template-part-2

**Comment by bparees** [comment]
lgtm

**Comment by jwforres** [comment]
@ahardin-rh FYI this content is accurate for 3.4 and later

**Comment by luciddreamz** [comment]
Awesome! Thanks @jwforres.

Do we have an example of service grouping for templates in the docs?

Do we want to suggest anything like: avoiding using the :latest tag for image streams referenced, setting memory and storage defaults, specific recommendations on what types of tags to use, or the fact that templates should work when deployed from the web console without the user needing to take further steps (like separately setting service accounts or secrets)? 

Up to you guys. Here's an example of a before and after I had done on Red Hat SSO 7 that might be helpful in this discussion: https://www.diffchecker.com/vWXxFgJc

**Comment by jwforres** [comment]
@luciddreamz @bparees what about something like:

template-other

@ahardin-rh is there a way to get that example block to indent under the bullet?

**Comment by luciddreamz** [comment]
@jwforres looks good

**Comment by bparees** [comment]
lgtm

**Comment by jwforres** [comment]
OK I will push those changes up in a bit

**Comment by ahardin-rh** [comment]
@jwforres you can place a + on the line between the bullet point and example to keep the example in line.

**Comment by jwforres** [comment]
ok updated to include the other recommendations section

**Comment by jwforres** [comment]
@ahardin-rh anything holding this one up?

**Comment by ahardin-rh** [comment]
Thanks! Sorry for the delay. I will apply some minor style edits in a separate PR.

File: dev_guide/templates.adoc [hunk 1]
```diff
@@ -339,20 +339,31 @@ metadata:
   name: "cakephp-mysql-example" <1>
   annotations:
     openshift.io/display-name: "CakePHP MySQL Example" <2>
-    description: "An example CakePHP application with a MySQL database" <3>
+    description: "An example CakePHP application with a MySQL database.\n\nFor more information see https://github.com/openshift/cakephp-ex" <3>
```
Comment (ahardin-rh) [inline dev_guide/templates.adoc]: @jwforres Is the \n\n supposed to be in there? Just checking as I make follow-up edits. Thanks!

Comment (ahardin-rh) [inline dev_guide/templates.adoc]: @jwforres Oh nevermind. I see now that it's new line.

**Comment by jwforres** [comment]
Yep trying to show how they can have multiple paragraphs

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/templates.adoc#dev-guide-templates[Templates]
|Added details about template writing for the best user experience.
%

**Pull request merged**